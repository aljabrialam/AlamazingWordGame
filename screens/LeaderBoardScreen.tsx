import React from 'react';

import {Text, View, Alert, TouchableOpacity} from 'react-native';
import RadioButton from '../components/RadioButton';
import styles from '../styles';
import AsyncStorage from '@react-native-async-storage/async-storage';

interface IProps {
  navigation: any;
}

export default class LeaderBoardScreen extends React.Component<IProps> {
  static navigationOptions = {
    title: '',
    headerStyle: {
      elevation: 0,
      shadowOpacity: 0,
      borderBottomWidth: 0,
    },
  };

  data = [
    {
      label: 'Cities',
      value: 'Cities',
    },
    {
      label: 'Foods',
      value: 'Foods',
    },
    {
      label: 'Animals',
      value: 'Animals',
    },
  ];

  state = {
    radio: '',
    score: '',
  };

  componentDidMount() {}

  asyncOperationOnAsyncStorage = async () => {
    await AsyncStorage.getItem('score').then(value => {
      this.setState({score: value});
    });
  };

  handleRadioButton = (value: any) => {
    this.setState({radio: value});
  };

  render() {
    const {navigate} = this.props.navigation;

    this.asyncOperationOnAsyncStorage();

    return (
      <View style={styles.homeContainer}>
        <Text style={styles.pageCountText}>1/1</Text>
        <View style={styles.gameTitleView}>
          <View style={styles.puzzleTitle}>
            <Text style={[styles.gameTitle, {margin: 10}]}>Words Puzzle</Text>
          </View>
        </View>

        <View style={styles.buttonContainer}>
          <RadioButton
            data={this.data}
            onSelect={this.handleRadioButton}
            styling={{color: 'blue', size: 30}}
          />
        </View>

        <View style={styles.startButtonContainer}>
          <TouchableOpacity
            style={styles.gameTitleView}
            hitSlop={{top: 30, bottom: 30, left: 30, right: 30}}
            onPress={() => {
              if (this.state.radio === '') {
                Alert.alert(
                  'To Start',
                  'Please select a category',
                  [{text: 'OK', onPress: () => {}}],
                  {cancelable: false},
                );
              } else {
                navigate('GameRoute', {radio: this.state.radio});
              }
            }}>
            <Text style={{marginTop: 8, fontSize: 20}}>START</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.buttonContainer}>
          <Text style={styles.gameTitle}>
            Leaders Board: {this.state.score}
          </Text>
        </View>
      </View>
    );
  }
}
