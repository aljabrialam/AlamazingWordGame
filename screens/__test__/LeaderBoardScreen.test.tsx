import React from 'react';
import renderer from 'react-test-renderer';

import LeaderBoardScreen from '../LeaderBoardScreen';

jest.mock('@react-native-async-storage/async-storage', () => {
  let cache: any = {};
  return {
    setItem: jest.fn((key, value) => {
      return new Promise((resolve, reject) => {
        return typeof key !== 'string' || typeof value !== 'string'
          ? reject(new Error('key and value must be string'))
          : resolve((cache[key] = value));
      });
    }),
    getItem: jest.fn((key, _value) => {
      return new Promise(resolve => {
        return cache.hasOwnProperty(key) ? resolve(cache[key]) : resolve(null);
      });
    }),
    removeItem: jest.fn(key => {
      return new Promise((resolve, reject) => {
        return cache.hasOwnProperty(key)
          ? resolve(delete cache[key])
          : reject('No such key!');
      });
    }),
    clear: jest.fn(_key => {
      return new Promise((resolve, _reject) => resolve((cache = {})));
    }),
    getAllKeys: jest.fn(_key => {
      return new Promise((resolve, _reject) => resolve(Object.keys(cache)));
    }),
  };
});

const navigationPropsLanding = {
  navigation: {
    navigate: jest.fn(),
    addListener: jest.fn(),
  },
  route: {params: {radio: 'Cities'}},
};
describe('<LeaderBoardScreen />', () => {
  it('has 5 child', () => {
    const tree: any = renderer
      .create(<LeaderBoardScreen {...navigationPropsLanding} />)
      .toJSON();
    expect(tree.children.length).toBe(5);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<LeaderBoardScreen {...navigationPropsLanding} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
