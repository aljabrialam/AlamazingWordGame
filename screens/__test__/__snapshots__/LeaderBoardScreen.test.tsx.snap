// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeaderBoardScreen /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 20,
        "fontWeight": "500",
        "justifyContent": "flex-start",
        "padding": 10,
        "textAlign": "center",
        "width": "100%",
      }
    }
  >
    1/1
  </Text>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#D9E7FB",
          "borderColor": "#d3d6da",
          "borderWidth": 2,
          "justifyContent": "center",
          "marginTop": 120,
          "paddingBottom": 10,
          "paddingEnd": 30,
          "paddingStart": 30,
          "paddingTop": 10,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
            },
            {
              "margin": 10,
            },
          ]
        }
      >
        Words Puzzle
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginTop": 50,
      }
    }
  >
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#d3d6da",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 50,
            "marginTop": 10,
            "paddingEnd": 30,
            "paddingStart": 30,
            "width": 240,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 30,
              "left": 30,
              "right": 30,
              "top": 30,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "marginTop": 8,
              }
            }
          >
            Cities
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#d3d6da",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 50,
            "marginTop": 10,
            "paddingEnd": 30,
            "paddingStart": 30,
            "width": 240,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 30,
              "left": 30,
              "right": 30,
              "top": 30,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "marginTop": 8,
              }
            }
          >
            Foods
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#fff",
            "borderColor": "#d3d6da",
            "borderRadius": 10,
            "borderWidth": 2,
            "height": 50,
            "marginTop": 10,
            "paddingEnd": 30,
            "paddingStart": 30,
            "width": 240,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            {
              "bottom": 30,
              "left": 30,
              "right": 30,
              "top": 30,
            }
          }
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flexDirection": "row",
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              {
                "fontSize": 20,
                "marginTop": 8,
              }
            }
          >
            Animals
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FECBCA",
        "borderColor": "#d3d6da",
        "borderWidth": 2,
        "height": 50,
        "marginTop": 50,
        "paddingEnd": 30,
        "paddingStart": 30,
        "width": 240,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      hitSlop={
        {
          "bottom": 30,
          "left": 30,
          "right": 30,
          "top": 30,
        }
      }
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flexDirection": "row",
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 20,
            "marginTop": 8,
          }
        }
      >
        START
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flexDirection": "column",
        "justifyContent": "center",
        "marginTop": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 20,
        }
      }
    >
      Leaders Board: 
    </Text>
  </View>
</View>
`;
